src/Sylius/Behat/Page/Admin/Order/ShowPage.php

```diff
@@ -171,16 +178,30 @@ public function getTaxTotal(): string
 
     public function hasShippingCharge(string $shippingCharge): bool
     {
-        $shippingChargesText = $this->getElement('shipping_charges')->getText();
+        $shippingChargesText = substr($this->getElement('shipping_adjustment_name')->getText(), 0, -1) . ' ' . $this->getElement('shipping_charges')->getText();
 
         return stripos($shippingChargesText, $shippingCharge) !== false;
     }
 
-    public function getPromotionTotal(): string
+    public function getOrderPromotionTotal(): string
     {
-        $promotionTotalElement = $this->getElement('promotion_total');
+        /** @var NodeElement[] $rows */
+        $rows = $this->getElement('table')->findAll('css', 'tbody tr');
+
+        $orderPromotionTotal = 0;
+
+        foreach ($rows as $row) {
+            $unitOrderPromotion =  $row->find('css', 'td:nth-child(4)')->getText();
+            $quantity = $row->find('css', 'td:nth-child(6)')->getText();
+            $itemOrderPromotion = (float) trim(str_replace('-$', '', $unitOrderPromotion)) * $quantity;
+            $orderPromotionTotal += (int) ($itemOrderPromotion * 100);
+        }
+
+        if ($orderPromotionTotal > 0) {
```

Comment:
Could be simplified to
```suggestion
        return $this->getFormattedMoney($orderPromotionTotal > 0? -1 * $orderPromotionTotal: $orderPromotionTotal);
```

And then we could remove lines below